src/wasm-stack.h

```diff
@@ -141,7 +141,7 @@ class BinaryInstWriter : public OverriddenVisitor<BinaryInstWriter> {
 
   // Keeps track of the binary index of the scratch locals used to lower
   // tuple.extract.
-  std::map<Type, Index> scratchLocals;
+  std::unordered_map<Type, Index> scratchLocals;
```

Comment:
this must be sorted, we iterate on it and it affects the order of `localTypes` which is reflected in the binary IIANM